manager.go

```diff
@@ -2089,94 +2090,44 @@ func (m *Manager) getTracefsRegex() (*regexp.Regexp, error) {
 // Once the limit is reached, the kernel refuses to load new probes and throws a "no such device"
 // error. To prevent this, start by cleaning up the kprobe_events entries of previous managers that
 // are not running anymore.
-func (m *Manager) cleanupTracefs() error {
-	// build the pattern to look for in kprobe_events and uprobe_events
+func (m *Manager) cleanupTraceFS() error {
 	pattern, err := m.getTracefsRegex()
 	if err != nil {
-		return err
+		return fmt.Errorf("tracefs regex: %s", err)
```

Comment:
change the error message to follow error patterns and use wrapping: 
something like `fmt.Errorf("failed to get tracefs regex: %w", err)`

Reply:
I actually don't like that message pattern for two reasons:
1. We already know it is an error (since it is an `error` object), so saying it is an error in the message is superfluous. If/when this error is logged, seems like the right time to call it out as an error (via log level or the message itself).
2. It makes nested errors verbose and repetitive. For example, `failed to attach kprobe: failed to get tracefs regex: ...`.

Reply:
Using `%w` is only necessary if you need the ability to detect a sentinel error. Otherwise you are exposing implementation details. Using a regex here is an implementation choice, so I do not want to expose the error type to the consumers of the library. That way, we can change the implementation later without worrying about breaking anyone.

Reply:
regarding your first message: that actually makes more sense to me than seeing `tracefs regex: ...`

regarding your second message: using %w is a **proper**  pattern for go, except for a few specific cases (for example when you need to support very old go versions or you want to return a completely new error. 

If you are referring to the second case, then at least document why you are not following the common pattern.

instead of "%w is only necessary", it should be" "not using %w is only acceptable in specific cases..."

Reply:
From official [Go wiki](https://github.com/golang/go/wiki/ErrorValueFAQ#i-am-already-using-fmterrorf-with-v-or-s-to-provide-context-for-an-error-when-should-i-switch-to-w)

> So use %w if you want to expose the underlying error to your callers. Keep in mind that doing so may be exposing implementation detail that can constrain the evolution of your code. Callers can depend on the type and value of the error you're wrapping, so changing that error can now break them.

Reply:
The concerns here are slightly different since it is a library that exposes an API, rather than an application which is internally consistent in the same repo.

Reply:
> that actually makes more sense to me than seeing `tracefs regex: ...`

That is your style preference, and mine is to add valuable context while being concise.